finance_manager/src/main/java/com/exadel/finance/manager/user/controller/UserController.java

```diff
@@ -0,0 +1,40 @@
+package com.exadel.finance.manager.user.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.exadel.finance.manager.user.model.entity.User;
+import com.exadel.finance.manager.user.model.entity.UserModel;
+import com.exadel.finance.manager.user.service.api.UserService;
+
+
+@RestController
+@RequiredArgsConstructor
+public class UserController
+{
+
+    private UserService userService;
+
+    @GetMapping("/profile")
+    public ResponseEntity<User> readUser() {
+        return new ResponseEntity<User>(userService.readUser(), HttpStatus.OK);
+    }
+
+    @PutMapping("/profile")
+    public ResponseEntity<User> updateUser(@RequestBody UserModel user) {
```

Comment:
parameters should be @RequestBody @Valid UserRequestDto user

@Valid is checking your fields are matching criteria (otherwise annotations like @NotNull has not effect)

Incoming object should be UserRequestDto

Outgoing object should be UserResponseDto